src/config/configJson.ts

```diff
@@ -87,56 +84,38 @@ export default class ConfigJsonConfigBackend {
 
 	private async write(): Promise<void> {
 		// writeToBoot uses fatrw to safely write to the boot partition
-		return hostUtils.writeToBoot(
-			await this.pathOnHost(),
-			JSON.stringify(this.cache),
-		);
+		return hostUtils.writeToBoot(await this.path(), JSON.stringify(this.cache));
 	}
 
 	private async read(): Promise<string> {
-		const filename = await this.pathOnHost();
+		const filename = await this.path();
 		return JSON.parse(await hostUtils.readFromBoot(filename, 'utf-8'));
 	}
 
-	private async resolveConfigPath(): Promise<string> {
+	private async path(): Promise<string> {
+		// TODO: Remove this once api-binder tests are migrated. The only
+		// time configPath is passed to the constructor is in the legacy tests.
 		if (this.configPath != null) {
 			return this.configPath;
 		}
-		if (constants.configJsonPathOnHost != null) {
-			return constants.configJsonPathOnHost;
+
+		// Prefer process.env.CONFIG_JSON_PATH if present for legacy support
+		if (constants.configJsonPathFromEnv != null) {
+			return constants.configJsonPathFromEnv;
 		}
 
-		const osVersion = await osRelease.getOSVersion(constants.hostOSVersionPath);
-		if (osVersion == null) {
+		// QUESTION: Do we still need this? Only versions <1.12 keep config.json at /mnt/cnf,
+		// and that's an ancient version. How far back do we need to support?
+		// See: https://github.com/balena-os/meta-balena/pull/251
+		const osSemver = await osRelease.getOSSemver(constants.hostOSVersionPath);
+		if (osSemver == null) {
 			throw new Error('Failed to detect OS version!');
 		}
-
-		if (/^(Resin OS|balenaOS)/.test(osVersion)) {
-			// In Resin OS 1.12, $BOOT_MOUNTPOINT was added and it coincides with config.json's path.
-			if (constants.bootMountPointFromEnv != null) {
-				return path.join(constants.bootMountPointFromEnv, 'config.json');
-			}
-			// Older 1.X versions have config.json here
+		if (semver.lte(osSemver, '1.12.0')) {
 			return '/mnt/conf/config.json';
```

Comment:
See QUESTION: do we need to be backwards compatible with OS versions older than 1.12?

Reply:
Oh, also OS 1.12.0 might still have `/mnt/boot` but they mount it to `/mnt/conf` in which case the mount script should still work and this case becomes useless

Reply:
I don't think we need to support this - I took a look at the oldest version listed when checking `balena os versions ${slug} | tail -1` for every device in `balena devices supported`, and the oldest is `v1.16.0` for intel-nuc and some other DTs.

Reply:
I removed the check for OS 1.12.0 entirely [here](https://github.com/balena-os/balena-supervisor/pull/2120/files#diff-3a034722503c6a7e6a3654f6f42c046256faaca32ff3d770ba5d727efff82470L114-L129)